src/ace/ari_text/parsemod.py

```diff
@@ -304,7 +311,8 @@ def p_structlist_end(p):
 
 def p_structpair(p):
     'structpair : VALSEG EQ VALSEG SC'
-    key = util.STRUCTKEY(p[1])
+    # Keys are case-insensitive so get folded to lower case
+    key = util.STRUCTKEY(p[1]).casefold()
```

Comment:
This does the case-folding (to lower in unicode-aware means) at the source rather than comparing options when looking up keys in this dict.